exercise/submission_models.py

```diff
@@ -96,7 +107,20 @@ def annotate_submitter_points(
                 forced_points=models.Max('grade', filter=models.Q(force_exercise_points=True)),
             )
             .annotate(**{
-                field_name: models.Case(*cases, default=models.Max('grade') if not force_zero else 0),
+                # Coalesce ensures that 0 is returned instead of None, if none
+                # of the submissions are in an expected status.
+                field_name: models.functions.Coalesce(
+                    models.Case(
+                        *cases,
+                        default=models.Max(
+                            'grade',
+                            filter=models.Q(
+                                status__in=statuses,
+                            ),
+                        ),
+                    ),
+                    0,
+                ) if not force_zero else models.Value(0),
```

Comment:
There was a bug in the previous version: the `if not force_zero else 0` expression applied to the `default` parameter of the `Case`. This means it only applied when the grading mode of the exercise was `BEST` (`default` corresponds to the `BEST` grading mode -- `LAST` and `force_exercise_points` are handled in the other cases). I moved the expression outside of the `Case`, so the entire case expression gets replaced with '0' if none of the exercises have been revealed.